src/MoBi.Core/Commands/SetParameterDimensionInBuildingBlockCommand.cs

```diff
@@ -39,17 +40,19 @@ protected override void ExecuteWith(IMoBiContext context)
          _parameter.Dimension = _newDimension;
          _parameter.DisplayUnit = _newDisplayUnit ?? displayUnitRetriever.PreferredUnitFor(_parameter);
 
-         updateFormulaDimension(_parameter.Formula, _newDimension, context);
-         updateFormulaDimension(_parameter.RHSFormula, context.DimensionFactory.RHSDimensionFor(_newDimension), context);
+         updateFormulaDimension(_parameter.Formula, () => _newDimension, context);
+
+         updateFormulaDimension(_parameter.RHSFormula, () => context.DimensionFactory.RHSDimensionFor(_newDimension), context);
```

Comment:
Well well. the RHS dimension was so far ALWAYS ADDED to the list if it was not available. Interesting side effects. now I have a lazy evaluation